### DISCUSSION TITLE
Import Error: Part matching query does not exist

### DISCUSSION THREAD

**Discussion by motusRP** [inventree/InvenTree#7065] [Q&A] [answered]
I'm trying to set up a new Inventree system and migrate some existing spreadsheet-based parts lists.  Working on 0.14.4.

So far i've been using Google Sheets to reformat our current parts lists into importable tables that i'm saving to CSV and then using the Admin pages to import in bulk.  I've been able to:
- import a Categories breakdown that matches our current system
- import Companies, and correctly flag them as Manufacturers and/or Suppliers
- import the Parts themselves

Now i'm trying to import a list of Manufacturer Parts, and while i've managed to create a CSV with the correct ID numbers for the part and manufacturer, when I try to import i'm getting this error:

![image](https://github.com/inventree/InvenTree/assets/54725635/ba21dc19-6db6-4de3-b2ec-eb9bcdad4579)

In this case the Part already exists in the Parts table (yes, the Name and IPN are the same):

![image](https://github.com/inventree/InvenTree/assets/54725635/20a4b1d1-9f47-48b1-b166-6a868edb28a7)

And the Company is in the Companies table as well:

![image](https://github.com/inventree/InvenTree/assets/54725635/e4687316-2dde-43f0-b3c0-b86d79f4320c)

I'm sure i'm doing something wrong here but I can't figure out what.  My import CSV for the Manufacturer Parts doesn't generate any errors (I corrected the ones it detected) this error about a part matching query is new.

Any help appreciated!

**Comment by SchrodingersGat** [comment, marked as answer]
@motusRP this feels like an issue with the underlying "parts" are not marked as "purchaseable". In the underlying model for the `ManufacturerPart` table, we have this:

![image](https://github.com/inventree/InvenTree/assets/10080325/ea5096e7-c568-4084-90ce-bee85a900aa7)

This will likely throw the (very unhelpful) error message you are seeing.

Can you confirm if the parts you are trying to link to are designed as "purchaseable"?

**Reply by motusRP** [reply]
Bingo!  That's it.  I knew I was missing something fundamental.  With that corrected, it imports correctly.  Thanks very much!

**Reply by HamzaF-m** [reply]
Hello MotusRP,

I'm having the same issue while migrating the data, where and what file i should make changes to, please advise?
